pubnub/lib/src/crypto/cryptorHeader.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:convert' show utf8;
+
+import 'package:pubnub/core.dart';
+
+/// @nodoc
+class CryptorHeader {
+  static const SENTINEL = 'PNED';
+  static const LEGACY_IDENTIFIER = '';
+  static const IDENTIFIER_LENGTH = 4;
+  static const MAX_VERSION = 1;
+
+  static CryptorHeaderV1? from(String id, List<int> metadata) {
+    if (id == LEGACY_IDENTIFIER) return null;
+    return CryptorHeaderV1(id, metadata.length);
+  }
+
+  static CryptorHeaderV1? tryParse(List<int> encryptedData) {
+    List<int> sentinel;
+    var version;
+    if (encryptedData.length >= 4) {
+      sentinel = encryptedData.sublist(0, 4).toList();
+      if (utf8.decode(sentinel, allowMalformed: true) != SENTINEL) return null;
+    }
+
+    if (encryptedData.length >= 5) {
+      version = encryptedData[4];
+    } else {
+      throw CryptoException('decryption error');
```

Comment:
Don't you want to provide more detailed messages that clearly indicate what's missing? For example, missing sentinel bytes, version byte, etc. Let's leave it as it is if you think it brings no value

Reply:
That would be helpful.
WIP: update along with tests in next commit.